datascience/tables.py

```diff
@@ -395,17 +395,156 @@ def append_column(self, label, values):
+            and ``values`` in addition to previous columns.
+
+        Raises:
+            ``ValueError``: If
+                - ``label`` is not a string.
+                - ``values`` is a list/array and does not have the same length
+                  as the number of rows in the table.
+
+        >>> table
+        letter | count | points
+        a      | 9     | 1
+        b      | 3     | 2
+        c      | 3     | 2
+        z      | 1     | 10
+
+        >>> table2 = table.append_column('new_col1', [10, 20, 30, 40])
+        >>> table3 = table2.append_column('new_col2', 'hello')
+        >>> table2
+        letter | count | points | new_col1
+        a      | 9     | 1      | 10
+        b      | 3     | 2      | 20
+        c      | 3     | 2      | 30
+        z      | 1     | 10     | 40
+        >>> table3
+        letter | count | points | new_col1 | new_col2
+        a      | 9     | 1      | 10       | hello
+        b      | 3     | 2      | 20       | hello
+        c      | 3     | 2      | 30       | hello
+        z      | 1     | 10     | 40       | hello
+        >>> table
+        letter | count | points
+        a      | 9     | 1
+        b      | 3     | 2
+        c      | 3     | 2
+        z      | 1     | 10
+
+        >>> table.append_column(123, [1, 2, 3, 4])
+        <ValueError raised>
+        >>> table.append_column('bad_col', [1, 2])
+        <ValueError raised>
+        """
+        new_table = self.copy()
+        new_table.append_column(label, values)
+        return new_table
+
+
     def relabel(self, column_label, new_label):
-        """Change the label of a column."""
-        assert column_label in self._columns
-        rewrite = lambda s: new_label if s == column_label else s
+        """Change the labels of columns specified by ``column_label`` to
+        labels in ``new_label``.
+
+        Args:
+            ``column_label`` (single str or list/array of str): The label(s) of
+                columns to be changed. Must be str.
+
+            ``new_label`` (single str or list/array of): The new label(s) of
+                columns to be changed. Must be str.
+
+                Number of elements must match number of elements in
+                ``column_label``.
+
+        Returns:
+            None
+
+        >>> table = Table([(1, 2, 3), (12345, 123, 5123)], ['points', 'id'])
+        >>> table.relabel('id', 'yolo')
+        >>> table
+        points | yolo
+        1      | 12345
+        2      | 123
+        3      | 5123
+        >>> table.relabel(['points', 'yolo'], ['red', 'blue'])
+        >>> table
+        red    | blue
+        1      | 12345
+        2      | 123
+        3      | 5123
+        """
+        if isinstance(column_label, str) and isinstance(new_label, str):
+            column_label, new_label = [column_label], [new_label]
+        d = dict(zip(column_label, new_label)) # dictioanry to map old labels to new ones
+        for col in d.keys():
```

Comment:
Let's call this dictionary `old_to_new` instead of `d` since it'll help someone who's reading this for the first time. Also, dictionaries iterate over their keys by default so you can more succinctly write `for col in old_to_new:`.

nit: You misspelled `dictioanry`!